src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -549,15 +549,19 @@ private async Task<bool> EnumerateTargetsAsync(TContext context)
                 readyToScanChannel.Writer.Complete();
             }
 
-            if (context.TargetsProvider.Skipped?.Count > 0)
+            if (context.TargetsProvider.Skipped != null)
             {
                 foreach (IEnumeratedArtifact artifact in context.TargetsProvider.Skipped)
                 {
-                    Notes.LogFileSkippedDueToSize(context, artifact.Uri.GetFilePath(), (long)artifact.SizeInBytes);
-                }
+                    if (artifact.SizeInBytes > 0)
+                    {
+                        _filesExceedingSizeLimitCount++;
+                        Notes.LogFileExceedingSizeLimitSkipped(context, artifact.Uri.GetFilePath(), (long)artifact.SizeInBytes);
+                        continue;
+                    }
 
-                //TBD resolve type mismatch
-                _ignoredFilesCount += (uint)context.TargetsProvider.Skipped.Count;
+                    Notes.LogEmptyFileSkipped(context, artifact.Uri.GetFilePath());
```

Comment:
>LogEmptyFileSkipped [](http://example.com/codeflow?start=26&length=19)

Note that we don't track/report out on skipping 0-byte files, though we do set a non-fatal runtime condition. How interesting is it that we don't scan a 0-byte file?

I suppose we still have a problem, though, there's at least a theoretical case that someone's analysis might be intended to flag all zero-byte files. :) We could allow this lower-bound threshold to be configurable and the core engine code would probably need to be updated.